src/main/java/org/codevillage/PutShapesChainEnd.java

```diff
@@ -0,0 +1,11 @@
+package org.codevillage;
+
+public class PutShapesChainEnd extends ShapeChainEnd{
+    @Override
+    public void construct() {
+        CanvasData canvasData = CanvasData.getInstance().getShapes();
```

Comment:
shouldn't this just be CanvasData.getInstance()?

Reply:
Actually I think it should be `List<Shape> canvasData = CanvasData.getInstance().getShapes()` since we are just modifying the array and not the entire singleton